src/components/Pricing/PricingSlider/pricingSliderLogic.ts

```diff
@@ -40,28 +40,38 @@ export const pricingSliderLogic = kea({
         finalCost: [
             (s) => [s.eventNumber, s.pricingOption],
             (eventNumber: number, pricingOption: PricingOptionType) => {
+                let finalCost = 0
+                let unitPricing = 0.000225
                 if (pricingOption === 'self-hosted') {
-                    let unitPricing = 0.000225
-
                     const estimatedCost = eventNumber * unitPricing
-                    let finalCost = estimatedCost > SCALE_MINIMUM_PRICING ? estimatedCost : SCALE_MINIMUM_PRICING
+                    finalCost = estimatedCost > SCALE_MINIMUM_PRICING ? estimatedCost : SCALE_MINIMUM_PRICING
 
                     if (eventNumber >= 10_000_000 && eventNumber < 100_000_000) {
                         unitPricing = 0.000045
-                        finalCost = 10_000_000 * 0.000225 + (eventNumber - 10_000_000) * 0.000045
-                    } else if (eventNumber >= 100000000) {
+                        finalCost = 10_000_000 * 0.000225 + (eventNumber - 10_000_000) * unitPricing
+                    } else if (eventNumber >= 100_000_000) {
                         unitPricing = 0.000009
                         finalCost =
-                            10_000_000 * 0.000225 + 90_000_000 * 0.000045 + (eventNumber - 100_000_000) * 0.000009
+                            10_000_000 * 0.000225 + 90_000_000 * 0.000045 + (eventNumber - 100_000_000) * unitPricing
                     }
 
                     actions.setAdditionalUnitPrice(unitPricing)
                     return Math.round(finalCost).toLocaleString()
                 }
 
                 // Cloud
-                const billableEvents = eventNumber - 1000000 > 0 ? eventNumber - 1000000 : 0
-                return Math.round(billableEvents * 0.000225).toLocaleString()
+                if (eventNumber > 1_000_000 && eventNumber <= 10_000_000) {
+                    unitPricing = 0.000225
+                    finalCost = (eventNumber - 1_000_000) * unitPricing
+                } else if (eventNumber > 10_000_000 && eventNumber <= 100_000_000) {
+                    unitPricing = 0.000075
+                    finalCost = 9_000_000 * 0.000225 + (eventNumber - 10_000_000) * unitPricing
+                } else if (eventNumber > 100_000_000) {
+                    unitPricing = 0.000075
+                    finalCost = 9_000_000 * 0.000225 + 90_000_000 * 0.000075 + (eventNumber - 100_000_000) * unitPricing
```

Comment:
Consider sth like this for it to be easier to update pricing/thresholds in the future and avoid typos (likely incorrect ts):
```suggestion
                threshold_to_price = [[1_000_000, 0], [10_000_000, 0.000225], [100_000_000, 0.000075], [Number.MAX_SAFE_INTEGER, 0.000025]]
                finalCost = 0
                already_counted_events = 0
                for ([threshold, unitPricing] of threshold_to_price) {
                    finalCost = finalCost + max(0, min(eventNumber - already_counted_events, threshold)) * unitPricing
                    already_counted_events = threshold
                }
```

Reply:
Thanks @tiina303! This makes it a lot better. @kpthatsme @jamesefhawkins ready for a final look before merging.

Reply:
maybe update the self-hosted one similarly?